src/controllers/blocks/BlocksExtrinsicsController.ts

```diff
@@ -0,0 +1,68 @@
+import { ApiPromise } from '@polkadot/api';
+import { RequestHandler } from 'express';
+
+import { BlocksService } from '../../services';
+import { INumberParam } from '../../types/requests';
+import AbstractController from '../AbstractController';
+
+interface ControllerOptions {
+	finalizes: boolean;
+}
+
+export default class BlocksExtrinsicsController extends AbstractController<BlocksService> {
+	constructor(api: ApiPromise, private readonly options: ControllerOptions) {
+		super(api, '/blocks/:blockId/extrinsics', new BlocksService(api));
+		this.initRoutes();
+	}
+
+	protected initRoutes(): void {
+		this.safeMountAsyncGetHandlers([
+			['/:extrinsicsIndex', this.getExtrinsicsByExtrsinsicIndex],
+		]);
+	}
+
+	/**
+	 *
+	 * @param _req Express Request
+	 * @param res Express Response
+	 */
+	private getExtrinsicsByExtrsinsicIndex: RequestHandler<INumberParam> = async (
+		{
+			params: { blockId, extrinsicsIndex },
+			query: { eventDocs, extrinsicDocs },
+		},
+		res
+	): Promise<void> => {
+		const hash = await this.getHashForBlock(blockId);
+
+		const eventDocsArg = eventDocs === 'true';
+		const extrinsicDocsArg = extrinsicDocs === 'true';
+
+		const queryFinalizedHead = !this.options.finalizes ? false : true;
+		const omitFinalizedTag = !this.options.finalizes ? true : false;
+
+		const options = {
+			eventDocs: eventDocsArg,
+			extrinsicDocs: extrinsicDocsArg,
+			checkFinalized: true,
+			queryFinalizedHead,
+			omitFinalizedTag,
```

Comment:
I think we should hardcode `omitFinalizedTag` too `true` in order to avoid any of the `isFinalizedBlock` logic. Then both `checkFinalized` and `queryFinalizedHead` can be whatever since (I think) they are not used.

Reply:
Yea I agree. `omitFinalizedTag` and `queryFinalizedHead`  are not connected to the output of this route, so we definitely don't need to assume the `this.options` might be fulfilled.